dwave/embedding/polynomialembeddor.py

```diff
@@ -0,0 +1,1222 @@
+        vlines = [[(x0, y, 1, k) for y in y_range] for k in range(self.L)]
+        return list(filter(self._contains_line, vlines))
+
+    def maximum_hline_bundle(self, y0, x0, x1):
+        """Compute a maximum set of horizontal lines in the unit cells ``(x,y0)``
+        for :math:`x0 \leq x \leq x1`.
+
+        INPUTS:
+            y0,x0,x1: int
+
+        OUTPUT:
+            list of lists of qubits
+        """
+        x_range = range(x0, x1 + 1) if x0 < x1 else range(x0, x1 - 1, -1)
+        hlines = [[(x, y0, 0, k) for x in x_range] for k in range(self.L)]
+        return list(filter(self._contains_line, hlines))
+
+    def maximum_ell_bundle(self, ell):
+        """Return a maximum ell bundle in the rectangle bounded by
+
+            :math:`\{x0,x1\} \\times \{y0,y1\}`
+
+        with vertical component
+
+            :math:`(x0,y0) ... (x0,y1) = {x0} \\times \{y0,...,y1\}`
+
+        and horizontal component
+
+            :math:`(x0,y0) ... (x1,y0) = \{x0,...,x1\} \\times \{y0\}`.
+
+        Note that we don't require :math:`x0 \leq x1` or :math:`y0 \leq y1`.  We go
+        through some shenanigans so that the qubits we return
+        are all in a path.  A nice side-effect of this is that
+
+            >>> chains = maximum_ell_bundle(...)
+            >>> edges = [zip(path,path[:-1]) for path in chains]
+
+        where ``edges`` will be a list of lists of chain edges.
+
+        INPUTS::
+            ell: a tuple of 4 integers defining the ell, ``(x0, x1, y0, y1)``
+
+        OUTPUT::
+            chains: list of lists of qubits
+
+        Note: this function only to be called to construct a
+        native clique embedding *after* the block embedding has
+        been constructed.  Using this to evaluate the goodness
+        of an ell block will be slow.
+        """
+        (x0, x1, y0, y1) = ell
+        hlines = self.maximum_hline_bundle(y0, x0, x1)
+        vlines = self.maximum_vline_bundle(x0, y0, y1)
+
+        if self.random_bundles:
+            shuffle(hlines)
+            shuffle(vlines)
+
+        return [v + h for h, v in zip(hlines, vlines)]
+
+    def _combine_clique_scores(self, rscore, hbar, vbar):
+        """Computes the score of a partial native clique embedding given the score
+        attained by the already-placed ells, together with the ell block
+        defined by ``hbar = (y0,xmin,xmax)``, and  ``vbar = (x0,ymin,ymax)``.
+
+        In the plain :class:`eden_processor` class, this is simply the number of ells
+        contained in the partial native clique after adding the new ells.
+        """
+        (y0, xmin, xmax) = hbar
+        (x0, ymin, ymax) = vbar
+        if rscore is None:
+            rscore = 0
+        hscore = self.hline_score(y0, xmin, xmax)
+        vscore = self.vline_score(x0, ymin, ymax)
+        if vscore < hscore:
+            score = rscore + vscore
+        else:
+            score = rscore + hscore
+        return score
+
+    def maxCliqueWithRectangle(self, R, maxCWR):
```

Comment:
Is this a special function? Otherwise, perhaps we should continue using snake case?